codec-http2/src/main/java/io/netty/handler/codec/http2/Http2MultiplexCodec.java

```diff
@@ -886,9 +886,10 @@ public void close(ChannelPromise promise) {
                         }
                     }
 
-                    // The promise should be notified before we call fireChannelInactive().
-                    promise.setSuccess();
+                    // The promise should be notified before we call fireChannelInactive(), and the closePromise should
+                    // be notified first for consistency with AbstractChannel so listeners of
```

Comment:
agreed it is 👍 ... however I plan on reverting this portion of the PR in favor of @normanmaurer's PR (see #7641 (comment)) so let me just leave this as is for now and it will go away after rebase.